src/components/Planning/ProjectProgrammedDialog/ProjectProgrammedDialog.tsx

```diff
@@ -0,0 +1,152 @@
+import { useState, MouseEvent, FC, useCallback } from 'react';
+import { Button } from 'hds-react/components/Button';
+import { Dialog } from 'hds-react/components/Dialog';
+import { useTranslation } from 'react-i18next';
+
+import { useAppDispatch, useAppSelector } from '@/hooks/common';
+import ProjectProgrammedSearch from './ProjectProgrammedSearch';
+import { IProgrammedProjectSuggestions } from '@/interfaces/searchInterfaces';
+import { IProjectsRequestObject } from '@/interfaces/projectInterfaces';
+import { silentPatchProjectsThunk } from '@/reducers/projectSlice';
+import { selectSelectedClass, selectSelectedSubClass } from '@/reducers/classSlice';
+import { useOptions } from '@/hooks/useOptions';
+import { current } from '@reduxjs/toolkit';
+
+interface IDialogProps {
+  handleClose: () => void;
+  isOpen: boolean;
+}
+
+const DialogContainer: FC<IDialogProps> = ({ isOpen, handleClose }) => {
```

Comment:
memoize the component because it is receiving properties

Reply:
done